roiextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -354,6 +365,23 @@ def data_generator(imaging):
                 metadata["Ophys"]["TwoPhotonSeries"][0],
                 dict(data=data, imaging_plane=imaging_plane),
             )
+            
+            if not use_times:
+                two_p_series_kwargs.update(
+                    starting_time=imaging.frame_to_time(0),
+                    rate=float(imaging.get_sampling_frequency())
+                )
+            else:
+                two_p_series_kwargs.update(
+                    timestamps=H5DataIO(
+                        imaging.frame_to_time(
+                            np.arange(imaging.get_num_frames())),
+                        compression="gzip"
+                    )
+                )
+                if "rate" in two_p_series_kwargs:
+                    del two_p_series_kwargs["rate"]
```

Comment:
Why add "rate" just to delete it?

Reply:
`rate` is added to the dictionary automatically when retrieving nwb metadata https://github.com/catalystneuro/roiextractors/blob/timestamps/roiextractors/extractors/nwbextractors/nwbextractors.py#L457-L460)

So in case `use_times` is used it's easier to just remove it IMO. If you prefer, we can avoid adding it in the `get_nwb_metadata` and take care of `rate` and/or `timestamps` directly in the `add_two_photon_series` function. What do you think?